### PR TITLE
Support Scala 2.12.0-M5

### PR DESCRIPTION
**Pull request by ktoso**
Review by @2m

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -36,6 +36,7 @@ object B extends Build {
         case "2.12.0-M2" => "2.2.5-M2"
         case "2.12.0-M3" => "2.2.5-M3"
         case "2.12.0-M4" => "2.2.6"
+        case "2.12.0-M5" => "3.0.0-RC4"
```
Comment (patriknw) [inline project/Build.scala]: have you heard anything of a scalatest 2.2.x version for M5? We will probably need that for Akka

**Comment by 2m** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

File: project/Build.scala (hunk 1 above)
Comment (ktoso) [inline project/Build.scala]: No, @bvenners only released 3.0.0 for 2.12.M5 so far.
Perhaps we could move on as well? Source code wise it should not be that painful (here in this repo no changes were required)

Comment (2m) [inline project/Build.scala]: Some discussion about it here: https://groups.google.com/forum/#!topic/scala-internals/DPCcIYR8-DE

File: plugin/src/test/scala/com/typesafe/genjavadoc/SignatureSpec.scala [hunk 2]
```diff
@@ -132,6 +139,7 @@ class SignatureSpec extends WordSpec with Matchers {
         c.getDeclaredMethods.filterNot(x ⇒ filter && (defaultFilteredStrings.exists { s => x.getName.contains(s) }
           || javaKeywords.contains(x.getName)
           || x.getName == "$init$" // These synthetic methods show up in 2.12.0-M4+ even though they are not in the generated Java sources
+          || (filter && Modifier.isStatic(x.getModifiers) && x.getName.endsWith("$")) // These synthetic static methods appear since 2.12.0-M5+ as companions to default methods
```
Comment (ktoso) [inline plugin/src/test/scala/com/typesafe/genjavadoc/SignatureSpec.scala]: I'm not entirely sure if this is correct though.
I'm filtering test results here, perhaps these methods should never appear in javadoc instead though? cc @2m 

I was not able to figure out where they come from though :(

Comment (2m) [inline plugin/src/test/scala/com/typesafe/genjavadoc/SignatureSpec.scala]: [Same approach](https://github.com/typesafehub/genjavadoc/commit/9705b7cec9c87f6841c4cda0b9e9b562280f2807) was taken when adding support for 2.12.0-M4. I think we can have it like this now, and when final scala 2.12.0 is out, we can filter on the output.

Comment (ktoso) [inline plugin/src/test/scala/com/typesafe/genjavadoc/SignatureSpec.scala]: Sounds like a good plan, thanks! 
Are you able to release this project so we unblock java8-compat?

**Pull request merged**